api/agent/call.go

```diff
@@ -48,14 +48,10 @@ type Param struct {
 }
 type Params []Param
 
-func FromRequest(appName, path string, req *http.Request) CallOpt {
+func FromRequest(app *models.App, path string, req *http.Request) CallOpt {
 	return func(a *agent, c *call) error {
-		app, err := a.da.GetApp(req.Context(), appName)
```

Comment:
so, this will make the cache pretty weird in hybrid. we want to keep the app lookup shoved down into the agent so that it can use the cache and either hit the db or call out to the hybrid api here. if we expect an app in this signature then we have to plumb that out from the agent (have a handle on DataAccessLayer in the caller) which isn't ideal. 

i.e. it seems we should have this be `a.da.GetAppByID` here and change `FromRequest` to be an id? i kind of get that in front of this we have to look up the app by name to get the id, the layering is definitely weird. mostly interested in your thoughts on layering wrt data access stuff